crates/shared/src/trade_finding/external.rs

```diff
@@ -186,19 +205,63 @@ mod dto {
         pub deadline: chrono::DateTime<chrono::Utc>,
     }
 
+    #[serde_as]
+    #[derive(Clone, Debug)]
+    pub enum Quote {
+        Legacy(LegacyQuote),
+        #[allow(unused)]
+        WithJitOrders(QuoteWithJITOrders),
+    }
+
+    impl<'de> Deserialize<'de> for Quote {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+        {
+            let value: serde_json::Value = Deserialize::deserialize(deserializer)?;
+
+            if value.get("clearingPrices").is_some()
+                || value.get("jitOrders").is_some()
+                || value.get("preInteractions").is_some()
+            {
+                Ok(Quote::WithJitOrders(
+                    QuoteWithJITOrders::deserialize(value).map_err(de::Error::custom)?,
+                ))
+            } else {
+                Ok(Quote::Legacy(
+                    LegacyQuote::deserialize(value).map_err(de::Error::custom)?,
+                ))
+            }
+        }
+    }
+
     #[serde_as]
     #[derive(Clone, Debug, Deserialize)]
     #[serde(rename_all = "camelCase")]
-    pub struct Quote {
+    pub struct LegacyQuote {
         #[serde_as(as = "HexOrDecimalU256")]
         pub amount: U256,
         pub interactions: Vec<Interaction>,
         pub solver: H160,
         pub gas: Option<u64>,
-        #[serde(default)]
         pub tx_origin: Option<H160>,
     }
 
+    #[serde_as]
+    #[derive(Clone, Debug, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    #[allow(unused)]
+    pub struct QuoteWithJITOrders {
```

Comment:
Same comment here that the future "regular" version should not have any qualifiers in the name.

Reply:
Renamed